### PR TITLE
Resolve immediately if no transitions defined

### PR DESCRIPTION
**Pull request by daun**
This change will help swup deal with a few common malconfigured cases:

- no animated elements were found → immediately resolve animation promise
- animated element has no CSS transitions → immediately resolve animation promise

The main goals here are:

- provide a happy path even if there's no animated element, defaulting to a `0s` timeout  
- reduce support burden by adding helpful error messages for the most common mistake

This will also fix:

- issue #444 by querying for animated elements below the `html` element, making sure the `from-*` classes added by swup won't be used as animation promises if the URL contains the word `/transition/`.

### CODE REVIEW

File: src/helpers/transitionEnd.js [hunk 1]
```diff
@@ -1,20 +1,9 @@
...
-	};
-
-	for (let name in transEndEventNames) {
-		if (el.style[name] !== undefined) {
-			return transEndEventNames[name];
-		}
+	if (window.ontransitionend === undefined && window.onwebkittransitionend !== undefined) {
+		return 'webkitTransitionEnd'
+	} else {
+		return 'transitionend'
```
Comment (gmrchk) [inline src/helpers/transitionEnd.js:5]: Not sure why is this change needed. Are the values different nowadays?

File: src/helpers/transitionProperty.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const transitionProperty = () => {
+	if (window.ontransitionend === undefined && window.onwebkittransitionend !== undefined) {
+		return 'WebkitTransition'
+	} else {
+		return 'transition'
+	}
```
Comment (gmrchk) [inline src/helpers/transitionProperty.js:6]: Looks like duplicate code. Code we reuse together with `transitionEnd` function?

File: src/modules/getAnimationPromises.js [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+	if (!animatedElements.length) {
+		console.error(`No animated elements found by selector ${this.options.animationSelector}`);
+		return [Promise.resolve()];
+	}
+
 	animatedElements.forEach((element) => {
+		const transitionDuration = window.getComputedStyle(element)[`${transitionProperty()}Duration`];
+		// Resolve immediately if no transition defined
+		if (!transitionDuration || transitionDuration == '0s') {
+			console.error(`No CSS transition defined for element of selector ${this.options.animationSelector}`);
```
Comment (gmrchk) [inline src/modules/getAnimationPromises.js:17]:
```suggestion
			console.error(`No CSS transition duration defined for element of selector ${this.options.animationSelector}`);
```

File: src/helpers/transitionEnd.js (hunk 1 above)
Comment (daun) [reply src/helpers/transitionEnd.js:5]: I thought the `opera` part looked dated, then just looked at how Vue currently handles transitions as it's baked into the core and works really well, and that's how they do it 🤠 We could revert this part to how it's done currently but this one is less code. However the opera part is not necessary anymore since they switched to Chromium as well. Want to revert it?

File: src/helpers/transitionProperty.js (hunk 2 above)
Comment (daun) [reply src/helpers/transitionProperty.js:6]: We could turn those two into one function that returns both? Or set a Webkit flag. Personally, I have a rule only to refactor and DRY for 3 or more occurrences of similar code so this one has flown under the radar 🤡

File: src/helpers/transitionEnd.js (hunk 1 above)
Comment (gmrchk) [reply src/helpers/transitionEnd.js:5]: Nah, just curious, let's roll with it!

**Review by gmrchk — APPROVED** [review]
